src/midi/midi_fluidsynth.cpp

```diff
@@ -69,6 +73,38 @@ void MidiHandlerFluidsynth::SetMixerLevel(const AudioFrame &desired_level) noexc
 	prescale_level.right = INT16_MAX * desired_level.right;
 }
 
+// Takes in a filename as a string with an optional post-fixed scaling percent,
+// removes it from the filename, and returns the percent as an integer.
+// If a percent isn't provided, then it returns the default_percent.
+int split_postfixed_percent(std::string &filename, const int default_percent)
```

Comment:
I really dislike having input/output parameters... Implementation is nice (bar some very unlikely edge cases), but it adds new code requiring `soundfont` string to be non-const…

A suggestion (not a requirement for this PR); this function could return `std::tuple<std::string, int>`, e.g.:
```C++
std::tuple<std::string, int> parse_sf_pref(const std::string &line,
                                           const int default_percent)
```
This way it could be used:
```C++
const auto sf_spec = parse_sf_pref(section->Get_string("soundfont"), 100);
auto soundfont = std::get<std::string>(sf_spec);
auto scale_by_percent = std::get<int>(sf_spec);
```

Mentioning it, because in the future we'll have structured binding in C++17, which will allow us to:

```C++
auto [soundfont, scale_by_percent] = parse_sf_pref(section->Get_string("soundfont"), 100);
```

Reply:
> I really dislike having input/output parameters.

So do it; the tuple result is exactly what this needs (thank you!), however the C++17 looks even sweeter still! It almost looks as elegant as Python :-)

Reply:
> Implementation is nice (bar some very unlikely edge cases)

Yes, I had some fun testing it with strange files names and it can indeed be fooled. 
For example:
1. Drop the SF2 extension: "something.sf2" :arrow_right: `"something"`
2. Rename the to end with space ### :arrow_right: `"something 150"`
3. Set the conf as such: `soundfont = something 150`

This will fool the algorithm into using the 150 as the percent and attempt to open "something" as the SF2 file. This will fail and the user will be informed that "something" could not be opened.